src/expr/src/scalar/func.rs

```diff
@@ -2945,6 +2965,7 @@ impl BinaryFunc {
             | DatePartTimestampTz
             | DateTruncTimestamp
             | DateTruncTimestampTz
+            | DateTruncInterval
```

Comment:
Nit: I'd insert in alphabetic order, here and below, but we're not enforcing it.